### PR TITLE
Update MQTT demo timeout and record length

### PR DESCRIPTION
**Pull request by moninom1**
Description
-----------
Update MQTT plain text demo timeout and record length to handle more packets. 

Test Steps
-----------
All demos run in Jenkins.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by AniruddhaKanhere — DISMISSED** [review]
@moninom1 can you please check whether this needs updating in other demos as well?

The change looks good!

**Review by aggarg — DISMISSED** [review]

**Review by ActoryOu — DISMISSED** [review]

**Review by sirnish — APPROVED** [review]
GTG

File: FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Plain_Text/DemoTasks/PlaintextMQTTExample.c [hunk 1]
```diff
@@ -182,14 +182,16 @@
 /**
  * @brief The length of the outgoing publish records array used by the coreMQTT
  * library to track QoS > 0 packet ACKS for outgoing publishes.
+ * Number of publishes = ulMaxPublishCount * mqttexampleTOPIC_COUNT
```
Comment (shubnil) [inline FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Plain_Text/DemoTasks/PlaintextMQTTExample.c:185]: Please add a comment explaining the relation between this count and the publish count "ulmaxpublishcount" with a note that a change in one will lead to a change in the other.

File: FreeRTOS-Plus/Demo/corePKCS11_MQTT_Mutual_Auth_Windows_Simulator/PKCS11_Mqtt_MutualAuthDemo.c [hunk 2]
```diff
@@ -161,14 +161,16 @@
...
  * library to track QoS > 0 packet ACKS for outgoing publishes.
+ * Number of publishes = ulMaxPublishCount * mqttexampleTOPIC_COUNT
  */
-#define mqttexampleOUTGOING_PUBLISH_RECORD_LEN            ( 10U )
+#define mqttexampleOUTGOING_PUBLISH_RECORD_LEN            ( 15U )
 
 /**
  * @brief The length of the incoming publish records array used by the coreMQTT
  * library to track QoS > 0 packet ACKS for incoming publishes.
+ * Number of publishes = ulMaxPublishCount * mqttexampleTOPIC_COUNT
```
Comment (shubnil) [inline FreeRTOS-Plus/Demo/corePKCS11_MQTT_Mutual_Auth_Windows_Simulator/PKCS11_Mqtt_MutualAuthDemo.c:171]: Please add a comment explaining the relation between this count and the publish count "ulmaxpublishcount" with a note that a change in one will lead to a change in the other.

**Review by shubnil — CHANGES_REQUESTED** [review]

File: FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_demo_helpers.c [hunk 3]
```diff
@@ -119,7 +119,7 @@
 /**
  * @brief Timeout for MQTT_ProcessLoop in milliseconds.
  */
-#define mqttexamplePROCESS_LOOP_TIMEOUT_MS           ( 500U )
+#define mqttexamplePROCESS_LOOP_TIMEOUT_MS           ( 2000U )
```
Comment (shubnil) [inline FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_demo_helpers.c:122]: How did we arrive at 2secs number. I understand that the previous value of 500ms was less. 
Ideally there should be a formula which co-relates the publish counts and mqttexamplePROCESS_LOOP_TIMEOUT_MS. And when we make the publish count as 15, we should get the above value.

Comment (moninom1) [reply FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_demo_helpers.c:122]: The arrival on 2sec number is based on periodically increasing mqttexamplePROCESS_LOOP_TIMEOUT_MS 
 and at what timeout we are able to keep up with the incoming ACKs and publishes.

Comment (shubnil) [reply FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_demo_helpers.c:122]: Please add this reasoning. After experimenting it has been found that for 15 publish messages, the apt timeout value is 2 secs. This should be updated if the number of publish messages change.

Comment (moninom1) [reply FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_demo_helpers.c:125]: Updated

File: FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_demo_helpers.c [hunk 4]
```diff
@@ -118,8 +118,11 @@
 
 /**
  * @brief Timeout for MQTT_ProcessLoop in milliseconds.
```
Comment (shubnil) [inline FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_demo_helpers.c:120]: Can we add one more line.
Currently, 15 topics are subscribed and based on experimentation a timeout of 2000ms is necessary to maintain a steady state throughout the demo.
So, the comment will look like this:

/**
 * @brief Timeout for MQTT_ProcessLoop in milliseconds.
 * This timeout corresponds to time taken by Number of topic subscribed to publish
 * messages from device to broker and broker to device to complete.
 * Currently, 15 topics are subscribed and based on experimentation a timeout of 2000ms is necessary
 * to maintain a steady state throughout the demo.
 * Any change in Number of topic subscribed needs updating mqttexamplePROCESS_LOOP_TIMEOUT_MS.
 */

File: FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_pkcs11_demo_helpers.c [hunk 5]
```diff
@@ -111,8 +111,11 @@
 
 /**
  * @brief Timeout for MQTT_ProcessLoop in milliseconds.
+ * This timeout corresponds to time taken by Number of topic subscribed to publish
+ * messages from device to broker and broker to device to complete.
+ * Any change in Number of topic subscribed needs updating mqttexamplePROCESS_LOOP_TIMEOUT_MS.
```
Comment (shubnil) [inline FreeRTOS-Plus/Demo/AWS/Mqtt_Demo_Helpers/mqtt_pkcs11_demo_helpers.c:116]: Please check once if we also have to update the mqttexampleOUTGOING_PUBLISH_RECORD_LEN in mqtt_pkcs11_demo_helpers.c and mqtt_demo_helpers.c

**Review by shubnil — DISMISSED** [review]
Thanks. The change looks good now.

**Review by ActoryOu — APPROVED** [review]

**Review by shubnil — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**